text/0000-cli-ui.md

```diff
@@ -0,0 +1,96 @@
+- Feature Name: cli-ui-improvements
+- Start Date: 2019-02-25
+- RFC PR: (leave this empty)
+- Notion Issue: (leave this empty)
+
+# Summary
+[summary]: #summary
+
+Transition from [Docopt] to [StructOpt], for an easier path to nice user experience (and better developer experience when working on) the command line interface.
+
+[Docopt]: https://docs.rs/docopt/1.0.2/docopt/
+[StructOpt]: https://docs.rs/structopt/0.2.14/structopt/
+
+# Motivation
+[motivation]: #motivation
+
+Our current command line parsing tool, [Docopt], works well but has several important limitations:
+
+- It does not readily support generating completions for any shell other than bash (and even these are a bit hairy).
+
+- It is, per its own description, a “black box”: if it works, it works well; if it doesn’t, developers are on their own for sorting out what is not parsing as expected and why. Those failures simply specify that the arguments failed to parse, not why or how, and they happen at runtime rather than compile time.
+
+- It requires custom wiring-up internally for every distinct subcommand to generate consistent help output, version output, etc., and custom parsing for each of those inputs.
+
+Switching to [StructOpt] improves our story in each of these areas:
+
+- It supports automatically generating completions for all major Unix shells and Powershell during the build step.
+
+- The configuration is validated by the compiler rather than at runtime, and the compiler’s error messages are usually much clearer about what is wrong than are the runtime errors from Doctopt.
+
+- It properly passes through global options to subcommands, and it understands all variants of “help” invocations (position and both flag styles) out of the box, and can be trivially configured to display help/usage information when no arguments are passed.
+
+Though StructOpt is not *universally* better than Docopt, it can generate similar output, but with a lower maintenance burden and a number of niceties *not* supported by Docopt.
+
+# Pedagogy
+[pedagogy]: #pedagogy
+
+This is purely an internal, developer-facing change, with *benefits* to the user but no changes in actual use of Notion. The documentation for StructOpt (and its core dependency [`clap`]) are sufficient for developers working on the project.
```

Comment:
I find this sentence confusing: it sounds like it's saying "this is user-visible but it's invisible to the user." Do you mean it does have user benefits but doesn't have any changes to workflow?

Reply:
Ah, great point. I'll update to that effect; thank you!